## Business Problem
The dataset is used to help the company understand customer churn by identifying behavioral and service-related patterns that indicate why customers leave or switch to another provider. The analysis also supports the development of effective retention strategies, such as targeted offers or service improvements, for customers who are at high risk of churning.


In [2]:
import pandas as pd

df = pd.read_csv('data/telecom_churn.csv')

print("=============================================")
print("ini adalah ringkasan informasi dari dataset ")
print("=============================================")
df.info()
df.shape
df.describe(include='all').T

ini adalah ringkasan informasi dari dataset 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3333 entries, 0 to 3332
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Churn            3333 non-null   int64  
 1   AccountWeeks     3333 non-null   int64  
 2   ContractRenewal  3333 non-null   int64  
 3   DataPlan         3333 non-null   int64  
 4   DataUsage        3333 non-null   float64
 5   CustServCalls    3333 non-null   int64  
 6   DayMins          3333 non-null   float64
 7   DayCalls         3333 non-null   int64  
 8   MonthlyCharge    3333 non-null   float64
 9   OverageFee       3333 non-null   float64
 10  RoamMins         3333 non-null   float64
dtypes: float64(5), int64(6)
memory usage: 286.6 KB


count        mean        std   min     25%     50%     75%  \
Churn            3333.0    0.144914   0.352067   0.0    0.00    0.00    0.00   
AccountWeeks     3333.0  101.064806  39.822106   1.0   74.00  101.00  127.00   
ContractRenewal  3333.0    0.903090   0.295879   0.0    1.00    1.00    1.00   
DataPlan         3333.0    0.276628   0.447398   0.0    0.00    0.00    1.00   
DataUsage        3333.0    0.816475   1.272668   0.0    0.00    0.00    1.78   
CustServCalls    3333.0    1.562856   1.315491   0.0    1.00    1.00    2.00   
DayMins          3333.0  179.775098  54.467389   0.0  143.70  179.40  216.40   
DayCalls         3333.0  100.435644  20.069084   0.0   87.00  101.00  114.00   
MonthlyCharge    3333.0   56.305161  16.426032  14.0   45.00   53.50   66.20   
OverageFee       3333.0   10.051488   2.535712   0.0    8.33   10.07   11.77   
RoamMins         3333.0   10.237294   2.791840   0.0    8.50   10.30   12.10   

                    max  
Churn              1.00  
AccountWeeks     243.00  
ContractRenewal    1.00  
DataPlan           1.00  
DataUsage          5.40  
CustServCalls      9.00  
DayMins          350.80  
DayCalls         165.00  
MonthlyCharge    111.30  
OverageFee        18.19  
RoamMins          20.00

In [3]:
df.head()

Churn  AccountWeeks  ContractRenewal  DataPlan  DataUsage  CustServCalls  \
0      0           128                1         1        2.7              1   
1      0           107                1         1        3.7              1   
2      0           137                1         0        0.0              0   
3      0            84                0         0        0.0              2   
4      0            75                0         0        0.0              3   

   DayMins  DayCalls  MonthlyCharge  OverageFee  RoamMins  
0    265.1       110           89.0        9.87      10.0  
1    161.6       123           82.0        9.78      13.7  
2    243.4       114           52.0        6.06      12.2  
3    299.4        71           57.0        3.10       6.6  
4    166.7       113           41.0        7.42      10.1

## Data Understanding

### Dataset Overview
- The dataset contains 3333 rows and 11 columns.
- The target variable is `Churn`.
- Some categorical features indicating ContractRenewal, DataPlan, and Churn as target variabel
- Most features are numerical ,such as DataUsage and monthly charges show wide variation.

### Initial Observations

- There are no missing values across the dataset, indicating good data quality.
- The average number of customer service calls is relatively low (1–2 calls), although some customers contacted customer service up to 9 times, which may indicate potential dissatisfaction.
- Only around 27% of customers subscribe to a data plan, while most customers rely primarily on voice and SMS services.
- The overall churn rate is approximately 14.49%, meaning that about one in seven customers has churned.
